src/lib/pick-defined-values.ts

```diff
@@ -0,0 +1,42 @@
+import { isString } from '@guardian/libs';
+
+type ValidTargeting<T, K> = '' extends T
+	? never
+	: [] extends T
+	? never
+	: [''] extends T
+	? never
+	: T extends boolean
+	? never
+	: T extends NonNullable<T>
+	? K
+	: never;
+type DefinedKeys<T> = { [K in keyof T]-?: ValidTargeting<T[K], K> }[keyof T];
+type ObjectWithDefinedValues<T> = Pick<T, DefinedKeys<T>>;
```

Comment:
Wow 😎 

![5x3bzr](https://user-images.githubusercontent.com/7014230/145216194-18b7f1b4-0e34-4c72-b998-12259d9c70bb.jpg)

Reply:
🤣  Yeah I explored these ideas in 10% time… it’s definitely right on the edge of usefulness.

Reply:
wow is right.